### ISSUE TITLE
Feature request: 6lowpan GHC support (RFC7400)

### ISSUE THREAD

**Issue by gebart** [RIOT-OS/RIOT#5197]
Currently not implemented in the gnrc 6lowpan stack is the generic header compression (GHC) scheme as described in RFC 7400
http://tools.ietf.org/html/rfc7400

This is just a suggestion for a project in case anyone is looking for something to dig into.

**Comment by OlegHahm** [comment]
AFAIK @ReneHerthel wanted to work on this. I have a non-RIOT implementation of GHC in https://github.com/OlegHahm/ghc, but it's not based on the RFC, but on an older version of the draft and it's untested (not even my own code).

**Comment by kaspar030** [comment]
> I have a non-RIOT implementation of GHC in
> https://github.com/OlegHahm/ghc, but it's not based on the RFC, but on
> an older version of the draft and it's untested (not even my own code).

Sounds like fun! ;)

**Comment by OlegHahm** [comment]
Well, to be honest some students of @emmanuelsearch hacked this long time ago and just wanted to make sure that the code isn't lost, but I cannot even tell if it makes sense to take this as a basis.

**Comment by PeterKietzmann** [comment]
Nope, @ReneHerthel decided against implemention GHC but @jfischer-phytec-iot planned to implement it. @ReneHerthel is working on the implementation of elided checksums (part of NHC) in RIOT.

**Comment by OlegHahm** [comment]
Thanks for the clarification, @PeterKietzmann. Sorry for my bad memory.

**Comment by PeterKietzmann** [comment]
No worries. IIRC I didn't inform you about that decision :-).
